docs/guides/1238-split-plan-with-packages.md

```diff
@@ -0,0 +1,247 @@
+
+*Given the below Dagger plan importing a package and located at the root of your project:*
+
+```cue title="main.cue"
+package main
+
+import(
+    "github.com/foo/bar" // <= package imported, example not working yet
+)
+```
+
+*The package imported right above needs to follow the below structure in the `cue.mod/usr` folder:*
+
+```console
+$ tree cue.mod/usr
+cue.mod/usr
+└── github.com
+    └── foo
+        └── bar
+            └── foo.cue // name of file not important
+
+3 directories, 1 file
+```
+
+#### Inside the imported package
+
+```cue title="cue.mod/usr/github.com/foo/bar/foo.cue"
+package baz
+
+#test: "ok"
+```
+
+### Importing packages
+
+In the above example, our `foo.cue` file has a package name (`baz`) different than his parent directory (`bar`).
+
+This is done on purpose, as this is the main difference with modules: a module can only contain files whose package names are equal to the name of their parent directories. This obligation doesn't apply to packages.
+
+However, when importing a package with a different name than its parent directory, it is important to specify the package name:
+
+```cue title="anything.cue (working)"
+package toto
+
+import(
+    "github.com/foo/bar:baz" // <= package imported properly, as the `baz` package was specifically requested
+)
+```
+
+If the `foo.cue` file specified a package name equivalent to its parent dirname, it wouldn't be necessary to explicitely require it. Exactly as for the module method, the import path would be:
+
+```cue title="anything.cue (working)"
+package toto
+
+import(
+    "github.com/foo/bar" // <= package imported properly, when package name == dirname
+)
+```
+
+### Practice
+
+Let's put everything above into practice. We are going to do 2 things:
+
+1. Port the config from a module split to a package split
+2. Change the imports to prepare for the push on Github
+
+For the sake of the exercise, let's pretend that the github repo's URL on which we will push the package is `github.com/user/myrepo`
+
+- `cd` to the root of the `daggerTest` project
+
+```console
+cd daggerTest
+```
+
+- Remove the email address `info@foo.bar` from the `cue.mod/module.cue` file
+
+```console
+echo -n 'module: ""' > cue.mod/module.cue
+```
+
+- Create the `cue.mod/usr` subfolders
+
+As the packages needs to follow `github.com/user/myrepo`, we need to create all these folders:
+
+```console
+mkdir -p cue.mod/usr/github.com/user/myrepo
+```
+
+- Move or link our `foo` and `bar` folders to this pkg
+
+We could do that in two ways: either move these folders to the `cue.mod/usr/github.com/user/myrepo` directory OR just symlink the folder. We are going to use the second method, as it will enable us to keep these folders at the root of the project:
+
+```console
+$ ln -s "$(pwd)/foo" "$(pwd)/cue.mod/usr/github.com/user/myrepo/foo"
+$ ln -s "$(pwd)/bar" "$(pwd)/cue.mod/usr/github.com/user/myrepo/bar"
```

Comment:
I think this complicates things. For development, you should create a repo outside the dagger project, for the package. Already early on this should imprint that these are two separate things: package repo vs project repo. I've seen users confuse these two in the past.

```console
$ mkdir myrepo # git init/clone, etc
$ ln -s myrepo daggerTest/cue.mod/github.com/user/myrepo
```
In your IDE you can add different folders to the same workspace.

This is not just for starting a package, it's the pattern to develop one, even if it already exists and this can be clarified. Eventually I'd expect us to have a `dagger project edit github.com/user/myrepo` which does something like this behind the scenes, like in Python.

Reply:
Even if you don't actually have a project and just need one to develop a package (i.e., only used for development) then you should still just make one up (like with `daggerTest`) and use the same pattern. That's what I do in Python as well.